controllers/tykoasapidefinition_controller.go

```diff
@@ -449,6 +670,62 @@ func getAPIID(tykOASCrd *v1alpha1.TykOasApiDefinition, tykOASDoc string) (string
 	return val, nil
 }
 
+func deleteOasApi(
+	ctx context.Context,
+	tykOASCrd *v1alpha1.TykOasApiDefinition,
+) error {
+	if tykOASCrd == nil {
+		return nil
+	}
+
+	if tykOASCrd.Status.ID != "" {
+		if err := klient.Universal.TykOAS().Delete(ctx, tykOASCrd.Status.ID); err != nil {
+			return err
+		}
+	}
+
+	util.RemoveFinalizer(tykOASCrd, keys.TykOASFinalizerName)
+
+	return nil
+}
+
+func (r *TykOasApiDefinitionReconciler) reconcileDeletingBaseOasAPIVersions(
+	ctx context.Context,
+	log logr.Logger,
+	tykOASCrd *v1alpha1.TykOasApiDefinition,
+) error {
+	if tykOASCrd == nil {
+		return nil
+	}
+
+	if tykOASCrd.Spec.Versioning != nil && tykOASCrd.Spec.Versioning.Enabled {
+		for _, v := range tykOASCrd.Spec.Versioning.Versions {
+			var versionOasAPI v1alpha1.TykOasApiDefinition
+
+			if err := r.Get(ctx, types.NamespacedName{
+				Name:      v.TykOasApiDefinitionRef,
+				Namespace: v.Namespace,
+			}, &versionOasAPI); err != nil {
+				if k8serrors.IsNotFound(err) {
+					log.Info("failed to get versioned TykOasApiDefinition CR", "err", err.Error())
+
+					return nil
```

Comment:
Instead of returning you should continue to remove reference from versions.

Reply:
if an error ocuurs, it means that the api version won't be fetched, It will lead to an error if we proceed.

Reply:
I mean try to fetch next version.
As we are returning nil, the link between base API and other versions will never be deleted.